scripts/benchmark.ts

```diff
@@ -202,16 +205,35 @@ class BuildTool {
       logger.log(`stderr: ${data}`);
     });
 
-    const stopServer = function () {
-      if (child.pid) {
-        kill(child.pid);
-      }
-    };
-
     return new Promise<DevServerResult>((resolve, reject) => {
       let startTime: number | null = null;
       let bundlerPid: number | null = null;
-      let timeUsage: number | null = null;
+
+      const stopServer = async () => {
+        if (!bundlerPid) {
+          throw new Error('Bundler pid not found');
+        }
+
+        const t1 = new Date().valueOf();
+        // wait bundler write cache to disk
+        while (true) {
+          const info = await pidusage(bundlerPid);
+          // CPU idle or timeout
+          if (info === 0 || new Date().valueOf() - t1 > 5000) {
+            break;
+          }
+          await sleep(200);
+        }
```

Comment:
Small fix:

- `info === 0` should be `info.cpu === 0`
- replace `while (true)` with `while (Date.now() - start < TIMEOUT)`

```suggestion
        const start = Date.now();
        const TIMEOUT = 5000;
        const INTERVAL = 200;

        // wait bundler to finish writing cache to disk
        while (Date.now() - start < TIMEOUT) {
          const { cpu } = await pidusage(bundlerPid);
          // CPU idle
          if (cpu === 0) {
            break;
          }
          await sleep(INTERVAL);
        }
```